# Energy

In [77]:
using JLD, FileIO

In [82]:
f = File(format"JLD","../data/uci/energy")
data = JLD.load(f);
println("Input dimension: ", size(data["X"]))
println("Output dimension: ", size(data["y"]))

Input dimension: (768, 8)
Output dimension: (768, 2)


In [83]:
files = map(i -> "energy_$(i).csv", 1:5);

In [84]:
X = mapreduce(file -> readcsv(file), hcat, files)'

5×5 Array{Float64,2}:
 10.1809   3.1203   3.119    2.02348  2.3187 
  9.55063  3.05243  3.05173  2.11569  2.57756
  9.94643  3.41418  3.41419  2.15133  2.30241
  9.38786  3.33076  3.33045  2.03239  2.03221
  9.78676  3.3593   3.35922  1.94144  1.94145

header:
* mean
* LLS
* Ridge
* GP
* SPN-GP 

In [85]:
μ = mean(X, 1)
σ = std(X, 1) ./ sqrt(size(X, 1))

header = ["Mean", "LLS", "Ridge", "GP", "SPN-GP"]

for (i, method) in enumerate(header)
    println(method, " & \$", @sprintf("%0.2f", μ[i]), " \\pm ", @sprintf("%0.2f", σ[i]) ,"\$")
end

Mean & $9.77 \pm 0.14$
LLS & $3.26 \pm 0.07$
Ridge & $3.25 \pm 0.07$
GP & $2.05 \pm 0.04$
SPN-GP & $2.23 \pm 0.11$


Mean & $9.77 \pm 0.14$
LLS & $3.26 \pm 0.07$
Ridge & $3.25 \pm 0.07$
GP & $2.05 \pm 0.04$
SPN-GP & $2.07 \pm 0.04$

# Concrete

In [90]:
f = File(format"JLD","../data/uci/concrete")
data = JLD.load(f);
println("Input dimension: ", size(data["X"]))
println("Output dimension: ", size(data["y"]))

Input dimension: (1030, 8)
Output dimension: (1030, 1)


In [91]:
files = map(i -> "concrete_$(i).csv", 1:5);

In [92]:
X = mapreduce(file -> readcsv(file), hcat, files)'

5×5 Array{Float64,2}:
 16.4522   9.55099   9.55226  6.17692  6.17644
 16.6221  10.7554   10.7566   6.38719  6.38748
 16.7283  10.386    10.3835   6.40419  6.39088
 16.2798  10.9258   10.9256   6.54563  6.55544
 16.1821  10.0504   10.0506   5.73197  5.72963

In [94]:
μ = mean(X, 1)
σ = std(X, 1) ./ sqrt(size(X, 1))

header = ["Mean", "LLS", "Ridge", "GP", "SPN-GP"]

for (i, method) in enumerate(header)
    println(method, " & \$", @sprintf("%0.3f", μ[i]), " \\pm ", @sprintf("%0.2f", σ[i]) ,"\$")
end

Mean & $16.453 \pm 0.10$
LLS & $10.334 \pm 0.25$
Ridge & $10.334 \pm 0.25$
GP & $6.249 \pm 0.14$
SPN-GP & $6.248 \pm 0.14$


Mean & $16.45 \pm 0.10$
LLS & $10.33 \pm 0.25$
Ridge & $10.33 \pm 0.25$
GP & $6.25 \pm 0.14$
SPN-GP & $6.27 \pm 0.20$

# CCPP

In [114]:
f = File(format"JLD","../data/uci/ccpp")
data = JLD.load(f);
println("Input dimension: ", size(data["X"]))
println("Output dimension: ", size(data["y"]))

Input dimension: (9568, 4)
Output dimension: (9568, 1)


In [115]:
files = map(i -> "ccpp_$(i).csv", 1:5);

In [116]:
X = mapreduce(file -> readcsv(file), hcat, files)'

5×5 Array{Float64,2}:
 17.1013  4.74048  4.74046  Inf  4.2355 
 16.9897  4.67634  4.67632  Inf  4.15439
 17.0008  4.64335  4.64335  Inf  4.11521
 17.0746  4.63464  4.63464  Inf  4.08591
 16.7967  4.47135  4.47134  Inf  3.96136

In [117]:
μ = mean(X, 1)
σ = std(X, 1) ./ sqrt(size(X, 1))

header = ["Mean", "LLS", "Ridge", "GP", "SPN-GP"]

for (i, method) in enumerate(header)
    println(method, " & \$", @sprintf("%0.2f", μ[i]), " \\pm ", @sprintf("%0.2f", σ[i]) ,"\$")
end

Mean & $16.99 \pm 0.05$
LLS & $4.63 \pm 0.04$
Ridge & $4.63 \pm 0.04$
GP & $Inf \pm NaN$
SPN-GP & $4.11 \pm 0.04$


In [118]:
10^2

100